### ISSUE TITLE
[Static Push] : File names with blank spaces cause problems

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#20295]
Reported via Support: [104010](https://dotcms.zendesk.com/agent/tickets/104010)
Reproducible in dotCMS 5.2.8.4, 5.3.8.4, and latest master.

Static pushing a File Asset with blank spaces in its name to an S3 AWS Environment is causing a different behavior compared to pushing a file **WITHOUT** any blank spaces.

**To Reproduce**
**NOTE: If the Demo Starter is not enough to reproduce this, please ask Jose Castro for details on the required data set.**
1. Create a test folder named `master-folder`, and upload a text file named `dotcms-test-file.txt` to it.
2. Set up an AWS S3 environment. You can ask Jose Castro for details on this as well.
3. Push the `dotcms-test-file.txt`. This works as expected. If you check the `dotcms.log`, you'll see this too:
```
INFO  bundlers.FileAssetBundler - +structuretype:4 +(path:/master-folder/dotcms-test-file.txt ) +(conhost:48190c8c-42c4-46af-8d1a-0cd5db894797  ) 
```
This is the Lucene query used to pull the file you selected.

4. Now, under the same `master-folder`, upload another text file named `dotcms test file`. Notice that the file has blank spaces, and no file extension.
5. Push this new file to S3. The Lucene looks like this now:
```
INFO  bundlers.FileAssetBundler - +structuretype:4 +(path:/master-folder/dotcms test file ) +(conhost:48190c8c-42c4-46af-8d1a-0cd5db894797  ) 
```
This query matches the file we selected, along with many other unrelated files in the repository. This is wrong.

6. Additionally, the push status is marked as `Successful` even though some of the pulled files actually failed to push.

**UPDATE on Apr 26th: Passing down a file with blank spaces is also causing the `HTMLPageAsContentBundler` to run and include a very high number of completely unrelated HTML Pages in the push.**

**Expected behavior**
Files with or without blank spaces in their names must be able to be static pushed correctly.

**Additional context**
In some customer environments, depending on the amount and size of files that are being pulled incorrectly, this is causing severe problems with the publishing queue

**Comment by jcastro-dotcms** [comment]
- PR for core: https://github.com/dotCMS/core/pull/20308
- PR for Enterprise: https://github.com/dotCMS/enterprise/pull/794

**Comment by jcastro-dotcms** [comment]
**Note to QA: Ticket #19646 is in charge of fixing important problems related to Static Publishing. Therefore, it must be merged and tested before testing this one.**

**Comment by nollymar** [comment]
**Passed internal QA:** I tested with several buckets and it is working as expected

**Comment by bryanboza** [comment]
Fixed, tested on release-21.06 // Postgres // FF

**Comment by jcastro-dotcms** [comment]
This commit did not referenced this ticket correctly, and it's required: https://github.com/dotCMS/enterprise/commit/8ab578db657f613bfc0a44485cb90b2cea22815f
